src/frontend/src/pages/GanttPage/GanttChartPage.tsx

```diff
@@ -165,13 +175,21 @@ const GanttChartPage: FC = () => {
 
   /***************************************************** */
 
-  const teamNameToGanttTasksMap = new Map<string, GanttTask[]>();
+  const addProjectHandler = (project: GanttTask) => {
+    setAddedProjects((prev) => [...prev, project]);
+  };
 
-  ganttProjectTasks.forEach((ganttTask) => {
-    const tasks: GanttTask[] = teamNameToGanttTasksMap.get(ganttTask.teamName) || [];
-    tasks.push(ganttTask);
-    teamNameToGanttTasksMap.set(ganttTask.teamName, tasks);
-  });
+  const addWorkPackageHandler = (workPackage: GanttTask) => {
+    setAddedWorkPackages((prev) => [...prev, workPackage]);
+  };
+
+  for (const project of ganttProjectTasks) {
+    const workPackagesToAdd = addedWorkPackages.filter(
+      (wp) => wp.projectId === project.id && project.workPackages.indexOf(wp) === -1
+    );
+    if (workPackagesToAdd.length === 0) continue;
+    project.workPackages.push(...workPackagesToAdd);
+  }
```

Comment:
some comments in the code u added here and above would be helpful. I understand the code in here before was pretty bad/ needs comments as well lol, apologies about that, was not expecting to be developing on it so soon